### PR TITLE
Keep track of viewed approaches

### PR DESCRIPTION
**Pull request by ErikSchierboom**
- Create table
- Add model for viewed exercise approach
- Add command to create viewed exercise approach
- Keep track of viewed approaches

### CODE REVIEW

File: app/models/user_track/viewed_exercise_approach.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class UserTrack::ViewedExerciseApproach < ApplicationRecord
+  belongs_to :user
+  belongs_to :track
+  belongs_to :exercise_approach, class_name: "Exercise::Approach"
```
Comment (iHiD) [inline app/models/user_track/viewed_exercise_approach.rb]:
```suggestion
  belongs_to :approach, class_name: "Exercise::Approach"
```

File: db/migrate/20231116111312_create_user_track_viewed_exercise_approach.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class CreateUserTrackViewedExerciseApproach < ActiveRecord::Migration[7.0]
+  def change
+    return if Rails.env.production?
+
+    create_table :user_track_viewed_exercise_approaches do |t|
+      t.references :user, null: false, foreign_key: true
+      t.references :track, null: false, foreign_key: true
+      t.references :exercise_approach, null: false, foreign_key: true, index: { name: , name: "index_user_track_viewed_exercise_approaches_on_approach_id" }
```
Comment (iHiD) [inline db/migrate/20231116111312_create_user_track_viewed_exercise_approach.rb]:
```suggestion
      t.references :approach, null: false, foreign_key: true, index: { name: , name: "index_user_track_viewed_exercise_approaches_on_approach_id" }
```

File: db/migrate/20231116111312_create_user_track_viewed_exercise_approach.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+      t.index %i[user_id track_id exercise_approach_id], unique: true, name: "index_user_track_viewed_exercise_approaches_uniq"
```
Comment (iHiD) [inline db/migrate/20231116111312_create_user_track_viewed_exercise_approach.rb]:
```suggestion
      t.index %i[user_id track_id approach_id], unique: true, name: "index_user_track_viewed_exercise_approaches_uniq"
```

File: test/commands/user_track/viewed_exercise_approach/create_test.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+  test "creates new record" do
+    user = create :user
+    track = create :track
+    exercise_approach = create(:exercise_approach, track:)
+
+    actual = UserTrack::ViewedExerciseApproach::Create.(user, track, exercise_approach)
+    assert_equal user, actual.user
+    assert_equal track, actual.track
+    assert_equal exercise_approach, actual.exercise_approach
```
Comment (iHiD) [inline test/commands/user_track/viewed_exercise_approach/create_test.rb]:
```suggestion
    assert_equal exercise_approach, actual.approach
```

**Review by iHiD — APPROVED** [review]
I'd rather just have `approach_id` and `approach` as the things within the model. I think that's enough context without the extra `exercise_`, but tell me if you decided against that for some reason :)

Otherwise, looks good!

**Review by iHiD — APPROVED** [review]
Nice work.

File: db/migrate/20231116111312_create_user_track_viewed_exercise_approach.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
+class CreateUserTrackViewedExerciseApproach < ActiveRecord::Migration[7.0]
+  def change
+    return if Rails.env.production?
+
+    create_table :user_track_viewed_exercise_approaches do |t|
+      t.references :user, null: false, foreign_key: true
+      t.references :track, null: false, foreign_key: true
+      t.references :approach, null: false, foreign_key: { to_table: :exercise_approaches }, index: { name: , name: "index_user_track_viewed_exercise_approaches_on_approach_id" }
```
Comment (ErikSchierboom) [inline db/migrate/20231116111312_create_user_track_viewed_exercise_approach.rb]:
```suggestion
      t.references :approach, null: false, foreign_key: { to_table: :exercise_approaches }, index: { name: "index_user_track_viewed_exercise_approaches_on_approach_id" }
```

**Pull request merged**